src/kvstore/global_kvs.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * Filename : global_kvs.c
+ * Description : 
+ * 
+ * Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ */
+
+#include <errno.h>
+#include <string.h>
+#include "kvstore.h"
+#include "global_kvs.h"
+
+/* Global definitions */
+kvs_idx_fid_t *g_fid = NULL;
+static struct kvstore *g_kvstore;
+struct kvs_idx g_index;
+extern bool kvs_init_done;
+
+/* Set an index mentioned by fid_str as global kvstore. Here it is assumed that
+ * the index is created */
+int nsal_set_global_kvs(const char *fid_str)
```

Comment:
Suggestion: We should probably have a function to get/return the fid of the global kvs ..probably `nsal_get_global_kvs ( )` 
Rationale : A user might want to set/get a kv value from global kvs, and its possible that the global kvs was not created/set.
The workflow will then be
- Get global kvs fid
- If the global kvs exists then set/get the required value.
Also g_fid should be static, it's generally a bad idea to use global variables without static. Infact, global/static g_fid could be avoided, and user can pass a local fid variable to nsal_get_global_kvs ( ) to get the global fid and then use this fid directly.

Alternatively, we can pack g_fid, g_index, g_kvstore in a structure, like g_kvs_handle. This could be a singleton object  and will be used internally by the set/get apis.
```
struct g_kvs_handle 
{
  struct kvstore g_kvstore;
 kvs_idx g_index
 kvs_idx_fid_t g_fid;
}

Reply:
Made the changes as per suggestion. Please review.